<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>R Package Introduction • dmdScheme</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="R Package Introduction">
<meta property="og:description" content="dmdScheme">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">dmdScheme</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.1.3.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/Howto_create_new_scheme.html">Howto Create a new scheme</a>
    </li>
    <li>
      <a href="../articles/minimum_requirements_dmdscheme.html">Minimum requirements for MetaData scheme based on dmdScheme</a>
    </li>
    <li>
      <a href="../articles/r_package_introduction.html">R Package Introduction</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/rkrug/dmdScheme/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="r_package_introduction_files/header-attrs-2.1/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>R Package Introduction</h1>
            <h3 class="subtitle">dmdScheme_0.9.5</h3>
                        <h4 class="author">Rainer M Krug <a href="mailto:Rainer.Krug@uzh.ch" class="email">Rainer.Krug@uzh.ch</a>
</h4>
            
            <h4 class="date">2020-04-30</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/rkrug/dmdScheme/blob/master/vignettes/r_package_introduction.Rmd"><code>vignettes/r_package_introduction.Rmd</code></a></small>
      <div class="hidden name"><code>r_package_introduction.Rmd</code></div>

    </div>

    
    
<div id="the-r-package" class="section level1">
<h1 class="hasAnchor">
<a href="#the-r-package" class="anchor"></a>The R Package</h1>
<p>The package <code>dmdScheme</code> is a base package for the usage and development of domain specific metadata schemes. It provides functionality to enter the metadata, validate the entered metadata, and export it to xml format for further processing by e.g. archival repositories. This document will give an overview over the <code>dmdScheme</code> package and what it contains. More detailed information about the development, including a suggested workflow to create a new scheme definition, can be found in the vignette <strong>Howto create a new scheme</strong>.</p>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>The recommended way is to install from CRAN the stable released version:</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"dmdScheme"</span>)</pre></body></html></div>
<p>To install the <strong>master</strong> branch, the stable branch which will become the new CRAN release, from the <a href="https://github.com/Exp-Micro-Ecol-Hub/dmdSchemeRepository/tree/master">dmdScheme repository on github</a>, run</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="co">## install the devtools package if not installed yet</span>
<span class="co"># install.packages("devtools")</span>

<span class="kw pkg">devtools</span><span class="kw ns">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span>(<span class="st">"Exp-Micro-Ecol-Hub/dmdScheme"</span>, <span class="kw">ref</span> <span class="kw">=</span> <span class="st">"master"</span>, <span class="kw">build_opts</span> <span class="kw">=</span> <span class="kw">NULL</span>)</pre></body></html></div>
<p>If you are feelinfg adventourous, want to live at the bleeding edge and can live with non-working features, you can install the <strong>dev</strong> branch. This branch is <strong>not stable</strong> and features and functionality can appear or be remioved without prior notice:</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="co">## install the devtools package if not installed yet</span>
<span class="co"># install.packages("devtools")</span>

<span class="kw pkg">devtools</span><span class="kw ns">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span>(<span class="st">"Exp-Micro-Ecol-Hub/dmdScheme"</span>, <span class="kw">ref</span> <span class="kw">=</span> <span class="st">"dev"</span>, <span class="kw">build_opts</span> <span class="kw">=</span> <span class="kw">NULL</span>)</pre></body></html></div>
<p>Other branches are not generally recommended for instalation unless you are developing <code>dmdScheme</code>!</p>
</div>
<div id="loading-the-package" class="section level2">
<h2 class="hasAnchor">
<a href="#loading-the-package" class="anchor"></a>Loading the package</h2>
<p>When you load the package, the definition of the scheme is downloaded from the <a href="https://github.com/Exp-Micro-Ecol-Hub/dmdSchemeRepository/tree/master">dmdScheme repository</a> installed to a temporary scheme library in a temporary directory for usage in this R session. As this scheme library is styored in a temporary directory, it will be deleted when youu quit your R session and re-downloaded each time you start a new session and load the package. To create a permanent package library you have to create a cache in the user directory. To do this, run</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="fu"><a href="../reference/cache.html">cache</a></span>(<span class="kw">createPermanent</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<p>and restart your R session. Noiw the definitions of the installed <code>dmdSchemes</code> will be installed in this user cache and be available permanently. For further info, see the documenatation of the command <code><a href="../reference/cache.html">cache()</a></code>.</p>
</div>
<div id="managing-installed-schemes" class="section level2">
<h2 class="hasAnchor">
<a href="#managing-installed-schemes" class="anchor"></a>Managing installed schemes</h2>
<p>Once loaded, the default scheme definition from the package is, when necessary, installed and used.</p>
<p>There are several commands to manage installed schemes. All these commands start with <code>scheme_</code>.</p>
<p>These are in detail:</p>
<div id="scheme_default" class="section level3">
<h3 class="hasAnchor">
<a href="#scheme_default" class="anchor"></a><code>scheme_default()</code>
</h3>
<p>returns the default scheme and version, i.e. the one with which the dmdScheme package is based. This can be the <code>dmdScheme</code> package itself, or a package crewated with <code><a href="../reference/make_new_package.html">make_new_package()</a></code>.</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="fu"><a href="../reference/scheme.html">scheme_default</a></span>()
<span class="co">#&gt;        name version</span>
<span class="co">#&gt; 1 dmdScheme   0.9.5</span></pre></body></html></div>
</div>
<div id="scheme_list_in_repo" class="section level3">
<h3 class="hasAnchor">
<a href="#scheme_list_in_repo" class="anchor"></a><code>scheme_list_in_repo()</code>
</h3>
<p>Show the schemes available in a repo. The defaut repo is <a href="https://github.com/Exp-Micro-Ecol-Hub/dmdSchemeRepository"></a>. The function reads and simly returns the file <code>SCHEME_DEFINITIONS.yaml</code> in the folder <code>schemes</code> in the repository.</p>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="fu"><a href="../reference/scheme.html">scheme_list_in_repo</a></span>()
<span class="co">#&gt; $dmdScheme_0.9.5</span>
<span class="co">#&gt; $dmdScheme_0.9.5$name</span>
<span class="co">#&gt; [1] "dmdScheme"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $dmdScheme_0.9.5$version</span>
<span class="co">#&gt; [1] "0.9.5"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $dmdScheme_0.9.5$description</span>
<span class="co">#&gt; [1] "Demo scheme from the dmdScheme package in R"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $emeScheme_0.9.5</span>
<span class="co">#&gt; $emeScheme_0.9.5$name</span>
<span class="co">#&gt; [1] "emeScheme"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $emeScheme_0.9.5$version</span>
<span class="co">#&gt; [1] "0.9.5"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $emeScheme_0.9.5$description</span>
<span class="co">#&gt; [1] "Scheme for Ecological Microcosom Experiments"</span></pre></body></html></div>
</div>
<div id="scheme_install" class="section level3">
<h3 class="hasAnchor">
<a href="#scheme_install" class="anchor"></a><code>scheme_install()</code>
</h3>
<p>Install a new scheme definition. In the normal usecase, this function uses <code>scheme_download_url()</code> to download the scheme defintion from the default <a href="https://github.com/Exp-Micro-Ecol-Hub/dmdSchemeRepository">github repository</a> and installs it. The usage is</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="fu"><a href="../reference/scheme.html">scheme_install</a></span>(<span class="kw">name</span> <span class="kw">=</span> <span class="st">"emeScheme"</span>, <span class="kw">version</span> <span class="kw">=</span> <span class="st">"0.9.5"</span>)
<span class="co">#&gt; Scheme definition `/var/folders/50/wcr5bjwn75q595n6x82gxj280000gq/T//RtmpXFEJYP/file556d24ef62a9/emeScheme_0.9.5.tar.gz, installed with</span>
<span class="co">#&gt; name:    emeScheme</span>
<span class="co">#&gt; version: 0.9.5</span></pre></body></html></div>
</div>
<div id="scheme_list" class="section level3">
<h3 class="hasAnchor">
<a href="#scheme_list" class="anchor"></a><code>scheme_list()</code>
</h3>
<p>Shows the installed schemes.</p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="fu"><a href="../reference/scheme.html">scheme_list</a></span>()
<span class="co">#&gt;        name version</span>
<span class="co">#&gt; 1 dmdScheme   0.9.5</span>
<span class="co">#&gt; 2 emeScheme   0.9.5</span></pre></body></html></div>
</div>
<div id="scheme_use" class="section level3">
<h3 class="hasAnchor">
<a href="#scheme_use" class="anchor"></a><code>scheme_use()</code>
</h3>
<p>Activate the scheme <code>NAME</code> with version <code>VERSION</code></p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="fu"><a href="../reference/scheme.html">scheme_use</a></span>(<span class="kw">name</span> <span class="kw">=</span> <span class="st">"emeScheme"</span>, <span class="kw">version</span> <span class="kw">=</span> <span class="st">"0.9.5"</span>)
<span class="co">#&gt; Theme switched to emeScheme_0.9.5</span></pre></body></html></div>
</div>
<div id="scheme_active" class="section level3">
<h3 class="hasAnchor">
<a href="#scheme_active" class="anchor"></a><code>scheme_active()</code>
</h3>
<p>Shows the currently active scheme.</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="fu"><a href="../reference/scheme.html">scheme_active</a></span>()
<span class="co">#&gt;        name version</span>
<span class="co">#&gt; 1 emeScheme   0.9.5</span>
<span class="fu"><a href="../reference/scheme.html">scheme_use</a></span>(<span class="kw">name</span> <span class="kw">=</span> <span class="st">"dmdScheme"</span>, <span class="kw">version</span> <span class="kw">=</span> <span class="st">"0.9.5"</span>)
<span class="co">#&gt; Theme switched to dmdScheme_0.9.5</span>
<span class="fu"><a href="../reference/scheme.html">scheme_active</a></span>()
<span class="co">#&gt;        name version</span>
<span class="co">#&gt; 1 dmdScheme   0.9.5</span></pre></body></html></div>
</div>
<div id="scheme_uninstall" class="section level3">
<h3 class="hasAnchor">
<a href="#scheme_uninstall" class="anchor"></a><code>scheme_uninstall()</code>
</h3>
<p>Uninstall an installed scheme. The scheme definition, is deleted from the scheme library, and moved to a temporary flder which wil be deleted at the end of the R session.</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="fu"><a href="../reference/scheme.html">scheme_uninstall</a></span>(<span class="kw">name</span> <span class="kw">=</span> <span class="st">"emeScheme"</span>, <span class="kw">version</span> <span class="kw">=</span> <span class="st">"0.9.5"</span>)
<span class="co">#&gt; Scheme emeScheme_0.9.5 deleted and moved to</span>
<span class="co">#&gt; [1] "/var/folders/50/wcr5bjwn75q595n6x82gxj280000gq/T//RtmpXFEJYP/file556d3b1d6dd3"</span></pre></body></html></div>
</div>
<div id="scheme_installed-returntrue-if-the-scheme-is-installed" class="section level3">
<h3 class="hasAnchor">
<a href="#scheme_installed-returntrue-if-the-scheme-is-installed" class="anchor"></a><code>scheme_installed() Return</code>TRUE, if the scheme is installed</h3>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="fu"><a href="../reference/scheme.html">scheme_installed</a></span>(<span class="kw">name</span> <span class="kw">=</span> <span class="st">"emeScheme"</span>, <span class="kw">version</span> <span class="kw">=</span> <span class="st">"0.9.5"</span>)
<span class="co">#&gt; [1] FALSE</span></pre></body></html></div>
</div>
<div id="scheme_path_xlsx" class="section level3">
<h3 class="hasAnchor">
<a href="#scheme_path_xlsx" class="anchor"></a><code>scheme_path_xlsx()</code>
</h3>
<p>Returns the path to the <code>.xlsx</code> representation of the currently active scheme.</p>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="fu"><a href="../reference/scheme_path_xlsx.html">scheme_path_xlsx</a></span>()
<span class="co">#&gt; [1] "/var/folders/50/wcr5bjwn75q595n6x82gxj280000gq/T//RtmpXFEJYP/dmdScheme_556d4df427cb/installedSchemes/dmdScheme_0.9.5/dmdScheme_0.9.5.xlsx"</span></pre></body></html></div>
</div>
<div id="scheme_path_xml" class="section level3">
<h3 class="hasAnchor">
<a href="#scheme_path_xml" class="anchor"></a><code>scheme_path_xml()</code>
</h3>
<p>Returns the path to the <code>.xml</code> representation of the currently active scheme.</p>
<div class="sourceCode" id="cb14"><html><body><pre class="r"><span class="fu"><a href="../reference/scheme_path_xml.html">scheme_path_xml</a></span>()
<span class="co">#&gt; [1] "/var/folders/50/wcr5bjwn75q595n6x82gxj280000gq/T//RtmpXFEJYP/dmdScheme_556d4df427cb/installedSchemes/dmdScheme_0.9.5/dmdScheme_0.9.5.xml"</span></pre></body></html></div>
</div>
<div id="scheme_make" class="section level3">
<h3 class="hasAnchor">
<a href="#scheme_make" class="anchor"></a><code>scheme_make()</code>
</h3>
<p>This function is used to package a new scheme. For details see the vignette <code>Howto Create a new scheme</code>.</p>
</div>
</div>
</div>
<div id="entering-new-metadata" class="section level1">
<h1 class="hasAnchor">
<a href="#entering-new-metadata" class="anchor"></a>Entering new Metadata</h1>
<p>To enter new data to the dmdScheme, you have to run the command</p>
<div class="sourceCode" id="cb15"><html><body><pre class="r"><span class="fu"><a href="../reference/open_new_spreadsheet.html">open_new_spreadsheet</a></span>()</pre></body></html></div>
<p>This will open Excel and the file should look similar to this, when looking at the second tab:</p>
<p><img src="figs/enter_new_data.png"></p>
<p>The following points are important to remember:</p>
<ol style="list-style-type: decimal">
<li>The file is saved in a <strong>temporary</strong> directory. It needs to be saved at a different location, if you want to keep the changes.</li>
<li>Data can only be entered in the green cells with. All other cells are write protected.</li>
</ol>
<p>After entering the data, save it to a location for further processing.</p>
<div id="importing-data-from-excel-sheet" class="section level2">
<h2 class="hasAnchor">
<a href="#importing-data-from-excel-sheet" class="anchor"></a>Importing Data from Excel Sheet</h2>
<p>Next, you have to import the data entered in the Excel sheet into R. For simplicity, we use here a file included in the package. If you want to load your own file, replace <code><a href="../reference/scheme_path_xlsx.html">scheme_path_xlsx()</a></code> with the file name and path to that file.</p>
<div class="sourceCode" id="cb16"><html><body><pre class="r"><span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/read_excel.html">read_excel</a></span>(
      <span class="kw">file</span> <span class="kw">=</span> <span class="fu"><a href="../reference/scheme_path_xlsx.html">scheme_path_xlsx</a></span>(),
      <span class="kw">verbose</span> <span class="kw">=</span> <span class="fl">TRUE</span>
)
<span class="co">#&gt; propertySet : valueProperty</span>
<span class="co">#&gt; Transposing...</span>
<span class="co">#&gt; Set names...</span>
<span class="co">#&gt; Set attributes...</span>
<span class="co">#&gt; Apply types...</span>
<span class="co">#&gt;    Apply type 'character' to 'name'...</span>
<span class="co">#&gt;    Apply type 'character' to 'temperature'...</span>
<span class="co">#&gt;    Apply type 'character' to 'light'...</span>
<span class="co">#&gt;    Apply type 'character' to 'humidity'...</span>
<span class="co">#&gt;    Apply type 'character' to 'incubator'...</span>
<span class="co">#&gt;    Apply type 'character' to 'container'...</span>
<span class="co">#&gt;    Apply type 'numeric' to 'microcosmVolume'...</span>
<span class="co">#&gt;    Apply type 'character' to 'mediaType'...</span>
<span class="co">#&gt;    Apply type 'numeric' to 'mediaConcentration'...</span>
<span class="co">#&gt;    Apply type 'character' to 'cultureConditions'...</span>
<span class="co">#&gt;    Apply type 'character' to 'comunityType'...</span>
<span class="co">#&gt;    Apply type 'character' to 'mediaAdditions'...</span>
<span class="co">#&gt;    Apply type 'integer' to 'duration'...</span>
<span class="co">#&gt;    Apply type 'character' to 'comment'...</span>
<span class="co">#&gt; Set class...</span>
<span class="co">#&gt; Done</span>
<span class="co">#&gt; propertySet : Genus</span>
<span class="co">#&gt; Set names...</span>
<span class="co">#&gt; Set attributes...</span>
<span class="co">#&gt; Apply types...</span>
<span class="co">#&gt;    Apply type 'character' to 'speciesID'...</span>
<span class="co">#&gt;    Apply type 'character' to 'colour'...</span>
<span class="co">#&gt;    Apply type 'character' to 'density'...</span>
<span class="co">#&gt;    Apply type 'character' to 'functionalGroup'...</span>
<span class="co">#&gt;    Apply type 'character' to 'comment'...</span>
<span class="co">#&gt; Set class...</span>
<span class="co">#&gt; Done</span>
<span class="co">#&gt; propertySet : Treatments</span>
<span class="co">#&gt; Set names...</span>
<span class="co">#&gt; Set attributes...</span>
<span class="co">#&gt; Apply types...</span>
<span class="co">#&gt;    Apply type 'character' to 'treatmentID'...</span>
<span class="co">#&gt;    Apply type 'character' to 'treatmentLevelHeight'...</span>
<span class="co">#&gt;    Apply type 'character' to 'comment'...</span>
<span class="co">#&gt; Set class...</span>
<span class="co">#&gt; Done</span>
<span class="co">#&gt; propertySet : Measurement</span>
<span class="co">#&gt; Set names...</span>
<span class="co">#&gt; Set attributes...</span>
<span class="co">#&gt; Apply types...</span>
<span class="co">#&gt;    Apply type 'character' to 'measurementID'...</span>
<span class="co">#&gt;    Apply type 'character' to 'variable'...</span>
<span class="co">#&gt;    Apply type 'character' to 'method'...</span>
<span class="co">#&gt;    Apply type 'character' to 'unit'...</span>
<span class="co">#&gt;    Apply type 'character' to 'object'...</span>
<span class="co">#&gt;    Apply type 'integer' to 'noOfSamplesInTimeSeries'...</span>
<span class="co">#&gt;    Apply type 'numeric' to 'samplingVolume'...</span>
<span class="co">#&gt;    Apply type 'character' to 'dataExtractionID'...</span>
<span class="co">#&gt;    Apply type 'character' to 'measuredFrom'...</span>
<span class="co">#&gt;    Apply type 'character' to 'comment'...</span>
<span class="co">#&gt; Set class...</span>
<span class="co">#&gt; Done</span>
<span class="co">#&gt; propertySet : DataExtraction</span>
<span class="co">#&gt; Set names...</span>
<span class="co">#&gt; Set attributes...</span>
<span class="co">#&gt; Apply types...</span>
<span class="co">#&gt;    Apply type 'character' to 'dataExtractionID'...</span>
<span class="co">#&gt;    Apply type 'character' to 'method'...</span>
<span class="co">#&gt;    Apply type 'character' to 'parameter'...</span>
<span class="co">#&gt;    Apply type 'character' to 'value'...</span>
<span class="co">#&gt;    Apply type 'character' to 'comment'...</span>
<span class="co">#&gt; Set class...</span>
<span class="co">#&gt; Done</span>
<span class="co">#&gt; propertySet : DataFileMetaData</span>
<span class="co">#&gt; Set names...</span>
<span class="co">#&gt; Set attributes...</span>
<span class="co">#&gt; Apply types...</span>
<span class="co">#&gt;    Apply type 'character' to 'dataFileName'...</span>
<span class="co">#&gt;    Apply type 'character' to 'columnName'...</span>
<span class="co">#&gt;    Apply type 'character' to 'columnData'...</span>
<span class="co">#&gt;    Apply type 'character' to 'mappingColumn'...</span>
<span class="co">#&gt;    Apply type 'character' to 'type'...</span>
<span class="co">#&gt;    Apply type 'character' to 'description'...</span>
<span class="co">#&gt;    Apply type 'character' to 'comment'...</span>
<span class="co">#&gt; Set class...</span>
<span class="co">#&gt; Done</span></pre></body></html></div>
<p>The <code>verbose = TRUE</code> argument will produce messages which will show you what is happening and will help to identify problems.</p>
</div>
<div id="print-dmdscheme-data" class="section level2">
<h2 class="hasAnchor">
<a href="#print-dmdscheme-data" class="anchor"></a>Print dmdScheme Data</h2>
<p><code>dmdScheme</code> Data can be printed by using the <code><a href="https://rdrr.io/r/base/print.html">print()</a></code> function. The function has three arguments which control the printout:</p>
<ul>
<li>
<code>printAttr</code>: if <code>TRUE</code> (default) print the <strong>basic</strong> attributes prefixed with <code>A</code>
</li>
<li>
<code>printExAttr</code>: if <code>TRUE</code> print the <strong>all</strong> attributes prefixed with <code>X</code>
</li>
<li>
<code>printData</code>: if <code>TRUE</code> (default) print the data prefixed with <code>D</code>
</li>
</ul>
<div class="sourceCode" id="cb17"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(
  <span class="no">x</span>,
  <span class="kw">printAttr</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
  <span class="kw">printExtAttr</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
  <span class="kw">printData</span> <span class="kw">=</span> <span class="fl">FALSE</span>
)
<span class="co">#&gt;  dmdScheme - dmdSchemeSet</span>
<span class="co">#&gt;      Experiment - dmdSchemeData</span>
<span class="co">#&gt;      Genus - dmdSchemeData</span>
<span class="co">#&gt;      Treatments - dmdSchemeData</span>
<span class="co">#&gt;      Measurement - dmdSchemeData</span>
<span class="co">#&gt;      DataExtraction - dmdSchemeData</span>
<span class="co">#&gt;      DataFileMetaData - dmdSchemeData</span></pre></body></html></div>
<div class="sourceCode" id="cb18"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(
  <span class="no">x</span>,
  <span class="kw">printAttr</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
  <span class="kw">printExtAttr</span> <span class="kw">=</span> <span class="fl">FALSE</span>,
  <span class="kw">printData</span> <span class="kw">=</span> <span class="fl">FALSE</span>
)
<span class="co">#&gt;  dmdScheme - dmdSchemeSet</span>
<span class="co">#&gt;      Experiment - dmdSchemeData</span>
<span class="co">#&gt; A       Names : name | temperature | light | humidity | incubator | container | microcosmVolume | mediaType | mediaConcentration | cultureConditions | comunityType | mediaAdditions | duration | comment</span>
<span class="co">#&gt; A       Units : NA | NA | NA | NA | NA | NA | ml | NA | g/l | NA | NA | NA | days | NA</span>
<span class="co">#&gt; A       Type  : character | character | character | character | character | character | numeric | character | numeric | character | character | character | integer | character</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;      Genus - dmdSchemeData</span>
<span class="co">#&gt; A       Names : speciesID | colour | density | functionalGroup | comment</span>
<span class="co">#&gt; A       Units : NA | NA | cells / ml | NA | NA</span>
<span class="co">#&gt; A       Type  : character | character | character | character | character</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;      Treatments - dmdSchemeData</span>
<span class="co">#&gt; A       Names : treatmentID | treatmentLevelHeight | comment</span>
<span class="co">#&gt; A       Units : NA | NA | NA</span>
<span class="co">#&gt; A       Type  : character | character | character</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;      Measurement - dmdSchemeData</span>
<span class="co">#&gt; A       Names : measurementID | variable | method | unit | object | noOfSamplesInTimeSeries | samplingVolume | dataExtractionID | measuredFrom | comment</span>
<span class="co">#&gt; A       Units : NA | NA | NA | NA | NA | NA | ml | NA | NA | NA</span>
<span class="co">#&gt; A       Type  : character | character | character | character | character | integer | numeric | character | character | character</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;      DataExtraction - dmdSchemeData</span>
<span class="co">#&gt; A       Names : dataExtractionID | method | parameter | value | comment</span>
<span class="co">#&gt; A       Units : NA | NA | NA | NA | NA</span>
<span class="co">#&gt; A       Type  : character | character | character | character | character</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;      DataFileMetaData - dmdSchemeData</span>
<span class="co">#&gt; A       Names : dataFileName | columnName | columnData | mappingColumn | type | description | comment</span>
<span class="co">#&gt; A       Units : NA | NA | NA | NA | NA | NA | NA</span>
<span class="co">#&gt; A       Type  : character | character | character | character | character | character | character</span></pre></body></html></div>
<div class="sourceCode" id="cb19"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(
  <span class="no">x</span>,
  <span class="kw">printAttr</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
  <span class="kw">printExtAttr</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
  <span class="kw">printData</span> <span class="kw">=</span> <span class="fl">FALSE</span>
)
<span class="co">#&gt;  dmdScheme - dmdSchemeSet</span>
<span class="co">#&gt;      Experiment - dmdSchemeData</span>
<span class="co">#&gt; A       Names : name | temperature | light | humidity | incubator | container | microcosmVolume | mediaType | mediaConcentration | cultureConditions | comunityType | mediaAdditions | duration | comment</span>
<span class="co">#&gt; A       Units : NA | NA | NA | NA | NA | NA | ml | NA | g/l | NA | NA | NA | days | NA</span>
<span class="co">#&gt; A       Type  : character | character | character | character | character | character | numeric | character | numeric | character | character | character | integer | character</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; XA      propertyName: Experiment</span>
<span class="co">#&gt; XA      suggestedValues: NA | treatment, in degrees celsius, measurement | treatment,light, dark, cycle , e.g. 16:8 LD | treatment, relative humidity in % | none, bench | NA | NA | NA | NA | axenic, dirty, clean | treatment, single trophic level, multiple trophic level | NA | NA | NA</span>
<span class="co">#&gt; XA      Description: The name of the experiment. | Temperature used for all treatments. If different between treatments, use "treatment" and specify in the Treatment sheet. | Light used for all treatments. If different between treatments, use "treatment" and specify in the Treatment sheet. | Humidity used for all treatments. If different between treatments, use "treatment" and specify in the Treatment sheet. | What type of incubator is used. | What type of container is used. | Volume of the microcosm container. Not the volume of the culture medium! | NA | NA | Conditions of the cultures for all treatments. | Characterisation of the microbe community. | NA | Length of the experiment in days. This should only include the time in which the measurements were taken! | Additional features of the Experiment you want to provide</span>
<span class="co">#&gt; XA      row.names: 1</span>
<span class="co">#&gt; XA      class: dmdSchemeData | data.frame</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;      Genus - dmdSchemeData</span>
<span class="co">#&gt; A       Names : speciesID | colour | density | functionalGroup | comment</span>
<span class="co">#&gt; A       Units : NA | NA | cells / ml | NA | NA</span>
<span class="co">#&gt; A       Type  : character | character | character | character | character</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; XA      propertyName: Genus</span>
<span class="co">#&gt; XA      suggestedValues: NA | NA | treatment | bacteria, bacterivore, predator, phototroph | NA</span>
<span class="co">#&gt; XA      allowedValues: NA | NA | NA | NA | NA</span>
<span class="co">#&gt; XA      Description: Id of the species and strain. Each speciesId has to be unique. | Where the species was obtained from. | Initial density used for all treatments. If different between treatments, use "treatment" and specify in the Treatment sheet. | Functional group of the species. | NA</span>
<span class="co">#&gt; XA      row.names: 1 | 2</span>
<span class="co">#&gt; XA      class: dmdSchemeData | data.frame</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;      Treatments - dmdSchemeData</span>
<span class="co">#&gt; A       Names : treatmentID | treatmentLevelHeight | comment</span>
<span class="co">#&gt; A       Units : NA | NA | NA</span>
<span class="co">#&gt; A       Type  : character | character | character</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; XA      propertyName: Treatments</span>
<span class="co">#&gt; XA      suggestedValues: species, temperatur, light, initial density, comunity composition, densities, dispersal, viscosity, disturbance, communityType | value,  variable: freetext | NA</span>
<span class="co">#&gt; XA      allowedValues: NA | NA | NA</span>
<span class="co">#&gt; XA      Description: ID of the the treatment decribed in this a row. Each treatmentId can occur multiple times as it can contain multiple treatment levels. | The value of the parameter if the parameter is constant over time, or a description of the variability. If unit is speciesId, comma separated list of all species in the treatment. | NA</span>
<span class="co">#&gt; XA      row.names: 1 | 2 | 3 | 4 | 5</span>
<span class="co">#&gt; XA      class: dmdSchemeData | data.frame</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;      Measurement - dmdSchemeData</span>
<span class="co">#&gt; A       Names : measurementID | variable | method | unit | object | noOfSamplesInTimeSeries | samplingVolume | dataExtractionID | measuredFrom | comment</span>
<span class="co">#&gt; A       Units : NA | NA | NA | NA | NA | NA | ml | NA | NA | NA</span>
<span class="co">#&gt; A       Type  : character | character | character | character | character | integer | numeric | character | character | character</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; XA      propertyName: Measurement</span>
<span class="co">#&gt; XA      suggestedValues: NA | O2 concentration, video, manual count, abundance, DNA | presens Optode, microscopy | %, mmol, count | species, OUT, gene, community, particles | NA | NA | NA | NA | NA</span>
<span class="co">#&gt; XA      allowedValues: NA | NA | NA | NA | NA | NA | NA | NA | NA | NA</span>
<span class="co">#&gt; XA      Description: Id of the Measurement process. This includes methodology, variables . Each measurementId specifies one Measurement process and must be unique in this column. Should be in the mapping column in the DataFileMetaData tab. | The variable measured. | Name of the method used. | Unit of the measured variable | The object measured. E.g. species in the case of manual count, gene for genetic analysis, particle for particle counters. | Total number of all samples in the time series. | The sampling volumne. If e.g. atmosphere in container is sampled (oxygen measurements), than enter 0.</span>
<span class="co">#&gt; Please use NA if sampling volumne is variable. | as used in the sheet DataExtraction, column dataExtractionID | if measured from the experiment, raw, else the measurementId (first column) of the Measurement it is based on. | NA</span>
<span class="co">#&gt; XA      row.names: 1 | 2 | 3 | 4</span>
<span class="co">#&gt; XA      class: dmdSchemeData | data.frame</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;      DataExtraction - dmdSchemeData</span>
<span class="co">#&gt; A       Names : dataExtractionID | method | parameter | value | comment</span>
<span class="co">#&gt; A       Units : NA | NA | NA | NA | NA</span>
<span class="co">#&gt; A       Type  : character | character | character | character | character</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; XA      propertyName: DataExtraction</span>
<span class="co">#&gt; XA      suggestedValues: NA | bemovi x.y.z | NA | NA | NA</span>
<span class="co">#&gt; XA      allowedValues: NA | NA | NA | NA | NA</span>
<span class="co">#&gt; XA      Description: Name of the DataExtraction process. This includes methodology, variables . Each name specifies one extraction process and can occur multiple times in the case of multiple parameters in the analysis. | Method used for the DataExtraction process. If possible including version (in the case of R packages). | parameter in the analysis. Only needs to be specified if it varies from the default. | value of the parameter (you can enter a number or a word) | NA</span>
<span class="co">#&gt; XA      row.names: 1</span>
<span class="co">#&gt; XA      class: dmdSchemeData | data.frame</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;      DataFileMetaData - dmdSchemeData</span>
<span class="co">#&gt; A       Names : dataFileName | columnName | columnData | mappingColumn | type | description | comment</span>
<span class="co">#&gt; A       Units : NA | NA | NA | NA | NA | NA | NA</span>
<span class="co">#&gt; A       Type  : character | character | character | character | character | character | character</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; XA      propertyName: DataFileMetaData</span>
<span class="co">#&gt; XA      suggestedValues: NA | NA | NA | NA | NA | NA | NA</span>
<span class="co">#&gt; XA      allowedValues: NA | NA | ID, Treatment, Measurement, Species, other | NA | integer, numeric, character, logical, datetime, date, time | NA | NA</span>
<span class="co">#&gt; XA      Description: the name of the data set. | Name of column in the data file. </span>
<span class="co">#&gt; Each column in the data file needs to be documented!</span>
<span class="co">#&gt; or NA</span>
<span class="co">#&gt; if it is for the whole data file and not specified in the dataFileName | The type of the data in the column.</span>
<span class="co">#&gt; ID: ID field (unique ID of unit of replication);</span>
<span class="co">#&gt; Treatment: specifies treatment; </span>
<span class="co">#&gt; Measurement: contains measurements; </span>
<span class="co">#&gt; Species: contains species; </span>
<span class="co">#&gt; other: other type of data | columnData = Treatment: </span>
<span class="co">#&gt; treatmentID as in the Treatment tab; </span>
<span class="co">#&gt; columnData = Species: </span>
<span class="co">#&gt; treatmentID refering to species composition as in the Treatment tab</span>
<span class="co">#&gt; columnData = Measurement: </span>
<span class="co">#&gt; measurementID as in the Measurement tab; </span>
<span class="co">#&gt; otherwise:</span>
<span class="co">#&gt; NA | Type of the column. | if column contains measurement: General description. If type is datatime, date, or time, give the order of year month day hour minute second as e.g. ymdhms, ymd, or hms. (Do not give any other information, e.g. give nothing about how months are entered (e.g. number or name), or how years, months, day, etc are separated. | NA</span>
<span class="co">#&gt; XA      row.names: 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 13 | 14 | 15 | 16 | 17 | 18 | 19 | 20</span>
<span class="co">#&gt; XA      class: dmdSchemeData | data.frame</span></pre></body></html></div>
<div class="sourceCode" id="cb20"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(
  <span class="no">x</span>,
  <span class="kw">printAttr</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
  <span class="kw">printExtAttr</span> <span class="kw">=</span> <span class="fl">TRUE</span>,
  <span class="kw">printData</span> <span class="kw">=</span> <span class="fl">TRUE</span>
)
<span class="co">#&gt;  dmdScheme - dmdSchemeSet</span>
<span class="co">#&gt;      Experiment - dmdSchemeData</span>
<span class="co">#&gt; A       Names : name | temperature | light | humidity | incubator | container | microcosmVolume | mediaType | mediaConcentration | cultureConditions | comunityType | mediaAdditions | duration | comment</span>
<span class="co">#&gt; A       Units : NA | NA | NA | NA | NA | NA | ml | NA | g/l | NA | NA | NA | days | NA</span>
<span class="co">#&gt; A       Type  : character | character | character | character | character | character | numeric | character | numeric | character | character | character | integer | character</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; XA      propertyName: Experiment</span>
<span class="co">#&gt; XA      suggestedValues: NA | treatment, in degrees celsius, measurement | treatment,light, dark, cycle , e.g. 16:8 LD | treatment, relative humidity in % | none, bench | NA | NA | NA | NA | axenic, dirty, clean | treatment, single trophic level, multiple trophic level | NA | NA | NA</span>
<span class="co">#&gt; XA      Description: The name of the experiment. | Temperature used for all treatments. If different between treatments, use "treatment" and specify in the Treatment sheet. | Light used for all treatments. If different between treatments, use "treatment" and specify in the Treatment sheet. | Humidity used for all treatments. If different between treatments, use "treatment" and specify in the Treatment sheet. | What type of incubator is used. | What type of container is used. | Volume of the microcosm container. Not the volume of the culture medium! | NA | NA | Conditions of the cultures for all treatments. | Characterisation of the microbe community. | NA | Length of the experiment in days. This should only include the time in which the measurements were taken! | Additional features of the Experiment you want to provide</span>
<span class="co">#&gt; XA      row.names: 1</span>
<span class="co">#&gt; XA      class: dmdSchemeData | data.frame</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;                name temperature        light humidity      incubator</span>
<span class="co">#&gt; D       1 ASR-expt1          20 semi-ambient  ambient not given here</span>
<span class="co">#&gt;                                    container microcosmVolume mediaType</span>
<span class="co">#&gt; D       1 Duran type bottle, red lids, 250ml             100       PPM</span>
<span class="co">#&gt;           mediaConcentration cultureConditions      comunityType</span>
<span class="co">#&gt; D       1               0.55             dirty initially unknown</span>
<span class="co">#&gt;                                                                   mediaAdditions</span>
<span class="co">#&gt; D       1 Wheat seeds added on specific dates, see file wheat_seed_additions.csv</span>
<span class="co">#&gt;           duration comment</span>
<span class="co">#&gt; D       1      100    &lt;NA&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;      Genus - dmdSchemeData</span>
<span class="co">#&gt; A       Names : speciesID | colour | density | functionalGroup | comment</span>
<span class="co">#&gt; A       Units : NA | NA | cells / ml | NA | NA</span>
<span class="co">#&gt; A       Type  : character | character | character | character | character</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; XA      propertyName: Genus</span>
<span class="co">#&gt; XA      suggestedValues: NA | NA | treatment | bacteria, bacterivore, predator, phototroph | NA</span>
<span class="co">#&gt; XA      allowedValues: NA | NA | NA | NA | NA</span>
<span class="co">#&gt; XA      Description: Id of the species and strain. Each speciesId has to be unique. | Where the species was obtained from. | Initial density used for all treatments. If different between treatments, use "treatment" and specify in the Treatment sheet. | Functional group of the species. | NA</span>
<span class="co">#&gt; XA      row.names: 1 | 2</span>
<span class="co">#&gt; XA      class: dmdSchemeData | data.frame</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;           speciesID  colour density functionalGroup</span>
<span class="co">#&gt; D       1      tt_1    ATCC       1     bacterivore</span>
<span class="co">#&gt; D       2   unknown unknown    &lt;NA&gt;         unknown</span>
<span class="co">#&gt;                                                            comment</span>
<span class="co">#&gt; D       1 http://www.lgcstandards-atcc.org/products/all/30007.aspx</span>
<span class="co">#&gt; D       2                                                     &lt;NA&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;      Treatments - dmdSchemeData</span>
<span class="co">#&gt; A       Names : treatmentID | treatmentLevelHeight | comment</span>
<span class="co">#&gt; A       Units : NA | NA | NA</span>
<span class="co">#&gt; A       Type  : character | character | character</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; XA      propertyName: Treatments</span>
<span class="co">#&gt; XA      suggestedValues: species, temperatur, light, initial density, comunity composition, densities, dispersal, viscosity, disturbance, communityType | value,  variable: freetext | NA</span>
<span class="co">#&gt; XA      allowedValues: NA | NA | NA</span>
<span class="co">#&gt; XA      Description: ID of the the treatment decribed in this a row. Each treatmentId can occur multiple times as it can contain multiple treatment levels. | The value of the parameter if the parameter is constant over time, or a description of the variability. If unit is speciesId, comma separated list of all species in the treatment. | NA</span>
<span class="co">#&gt; XA      row.names: 1 | 2 | 3 | 4 | 5</span>
<span class="co">#&gt; XA      class: dmdSchemeData | data.frame</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;             treatmentID treatmentLevelHeight comment</span>
<span class="co">#&gt; D       1 Lid_treatment                Loose    &lt;NA&gt;</span>
<span class="co">#&gt; D       2 Lid_treatment                Tight    &lt;NA&gt;</span>
<span class="co">#&gt; D       3     species_1        tt_1, unknown    &lt;NA&gt;</span>
<span class="co">#&gt; D       4     species_2              unknown    &lt;NA&gt;</span>
<span class="co">#&gt; D       5     species_3                 tt_1    &lt;NA&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;      Measurement - dmdSchemeData</span>
<span class="co">#&gt; A       Names : measurementID | variable | method | unit | object | noOfSamplesInTimeSeries | samplingVolume | dataExtractionID | measuredFrom | comment</span>
<span class="co">#&gt; A       Units : NA | NA | NA | NA | NA | NA | ml | NA | NA | NA</span>
<span class="co">#&gt; A       Type  : character | character | character | character | character | integer | numeric | character | character | character</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; XA      propertyName: Measurement</span>
<span class="co">#&gt; XA      suggestedValues: NA | O2 concentration, video, manual count, abundance, DNA | presens Optode, microscopy | %, mmol, count | species, OUT, gene, community, particles | NA | NA | NA | NA | NA</span>
<span class="co">#&gt; XA      allowedValues: NA | NA | NA | NA | NA | NA | NA | NA | NA | NA</span>
<span class="co">#&gt; XA      Description: Id of the Measurement process. This includes methodology, variables . Each measurementId specifies one Measurement process and must be unique in this column. Should be in the mapping column in the DataFileMetaData tab. | The variable measured. | Name of the method used. | Unit of the measured variable | The object measured. E.g. species in the case of manual count, gene for genetic analysis, particle for particle counters. | Total number of all samples in the time series. | The sampling volumne. If e.g. atmosphere in container is sampled (oxygen measurements), than enter 0.</span>
<span class="co">#&gt; Please use NA if sampling volumne is variable. | as used in the sheet DataExtraction, column dataExtractionID | if measured from the experiment, raw, else the measurementId (first column) of the Measurement it is based on. | NA</span>
<span class="co">#&gt; XA      row.names: 1 | 2 | 3 | 4</span>
<span class="co">#&gt; XA      class: dmdSchemeData | data.frame</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;                  measurementID  variable         method               unit</span>
<span class="co">#&gt; D       1 oxygen concentration        DO presens Optode                  %</span>
<span class="co">#&gt; D       2            abundance abundance      molecular              count</span>
<span class="co">#&gt; D       3                smell     smell           nose rotten eggs or not</span>
<span class="co">#&gt; D       4         sequenceData       DNA            NGS         Nucleotide</span>
<span class="co">#&gt;                 object noOfSamplesInTimeSeries samplingVolume</span>
<span class="co">#&gt; D       1    community                      50            0.0</span>
<span class="co">#&gt; D       2      species                       6            0.5</span>
<span class="co">#&gt; D       3    community                       6            0.0</span>
<span class="co">#&gt; D       4 DNA fragment                       6            0.0</span>
<span class="co">#&gt;              dataExtractionID measuredFrom comment</span>
<span class="co">#&gt; D       1                none          raw    &lt;NA&gt;</span>
<span class="co">#&gt; D       2 Mol_Analy_pipeline1 sequenceData    &lt;NA&gt;</span>
<span class="co">#&gt; D       3                none          raw    &lt;NA&gt;</span>
<span class="co">#&gt; D       4                none          raw    &lt;NA&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;      DataExtraction - dmdSchemeData</span>
<span class="co">#&gt; A       Names : dataExtractionID | method | parameter | value | comment</span>
<span class="co">#&gt; A       Units : NA | NA | NA | NA | NA</span>
<span class="co">#&gt; A       Type  : character | character | character | character | character</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; XA      propertyName: DataExtraction</span>
<span class="co">#&gt; XA      suggestedValues: NA | bemovi x.y.z | NA | NA | NA</span>
<span class="co">#&gt; XA      allowedValues: NA | NA | NA | NA | NA</span>
<span class="co">#&gt; XA      Description: Name of the DataExtraction process. This includes methodology, variables . Each name specifies one extraction process and can occur multiple times in the case of multiple parameters in the analysis. | Method used for the DataExtraction process. If possible including version (in the case of R packages). | parameter in the analysis. Only needs to be specified if it varies from the default. | value of the parameter (you can enter a number or a word) | NA</span>
<span class="co">#&gt; XA      row.names: 1</span>
<span class="co">#&gt; XA      class: dmdSchemeData | data.frame</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;              dataExtractionID method parameter value</span>
<span class="co">#&gt; D       1 Mol_Analy_pipeline1   &lt;NA&gt;      &lt;NA&gt;  &lt;NA&gt;</span>
<span class="co">#&gt;                                   comment</span>
<span class="co">#&gt; D       1 See description in file xxx.yyy</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;      DataFileMetaData - dmdSchemeData</span>
<span class="co">#&gt; A       Names : dataFileName | columnName | columnData | mappingColumn | type | description | comment</span>
<span class="co">#&gt; A       Units : NA | NA | NA | NA | NA | NA | NA</span>
<span class="co">#&gt; A       Type  : character | character | character | character | character | character | character</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; XA      propertyName: DataFileMetaData</span>
<span class="co">#&gt; XA      suggestedValues: NA | NA | NA | NA | NA | NA | NA</span>
<span class="co">#&gt; XA      allowedValues: NA | NA | ID, Treatment, Measurement, Species, other | NA | integer, numeric, character, logical, datetime, date, time | NA | NA</span>
<span class="co">#&gt; XA      Description: the name of the data set. | Name of column in the data file. </span>
<span class="co">#&gt; Each column in the data file needs to be documented!</span>
<span class="co">#&gt; or NA</span>
<span class="co">#&gt; if it is for the whole data file and not specified in the dataFileName | The type of the data in the column.</span>
<span class="co">#&gt; ID: ID field (unique ID of unit of replication);</span>
<span class="co">#&gt; Treatment: specifies treatment; </span>
<span class="co">#&gt; Measurement: contains measurements; </span>
<span class="co">#&gt; Species: contains species; </span>
<span class="co">#&gt; other: other type of data | columnData = Treatment: </span>
<span class="co">#&gt; treatmentID as in the Treatment tab; </span>
<span class="co">#&gt; columnData = Species: </span>
<span class="co">#&gt; treatmentID refering to species composition as in the Treatment tab</span>
<span class="co">#&gt; columnData = Measurement: </span>
<span class="co">#&gt; measurementID as in the Measurement tab; </span>
<span class="co">#&gt; otherwise:</span>
<span class="co">#&gt; NA | Type of the column. | if column contains measurement: General description. If type is datatime, date, or time, give the order of year month day hour minute second as e.g. ymdhms, ymd, or hms. (Do not give any other information, e.g. give nothing about how months are entered (e.g. number or name), or how years, months, day, etc are separated. | NA</span>
<span class="co">#&gt; XA      row.names: 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 13 | 14 | 15 | 16 | 17 | 18 | 19 | 20</span>
<span class="co">#&gt; XA      class: dmdSchemeData | data.frame</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;                             dataFileName    columnName  columnData</span>
<span class="co">#&gt; D       1  dissolved_oxygen_measures.csv        Jar_ID          ID</span>
<span class="co">#&gt; D       2  dissolved_oxygen_measures.csv            DO Measurement</span>
<span class="co">#&gt; D       3  dissolved_oxygen_measures.csv        Unit_1       other</span>
<span class="co">#&gt; D       4  dissolved_oxygen_measures.csv          Mode       other</span>
<span class="co">#&gt; D       5  dissolved_oxygen_measures.csv      Location       other</span>
<span class="co">#&gt; D       6  dissolved_oxygen_measures.csv     Date_time       other</span>
<span class="co">#&gt; D       7  dissolved_oxygen_measures.csv Lid_treatment   Treatment</span>
<span class="co">#&gt; D       8  dissolved_oxygen_measures.csv      Jar_type       other</span>
<span class="co">#&gt; D       9  dissolved_oxygen_measures.csv        Jar_ID          ID</span>
<span class="co">#&gt; D       10                     smell.csv          &lt;NA&gt;     Species</span>
<span class="co">#&gt; D       11                     smell.csv         smell Measurement</span>
<span class="co">#&gt; D       12                     smell.csv          Date       other</span>
<span class="co">#&gt; D       13                     smell.csv Lid_treatment   Treatment</span>
<span class="co">#&gt; D       14                     smell.csv      Jar_type       other</span>
<span class="co">#&gt; D       15                abundances.csv          &lt;NA&gt;     Species</span>
<span class="co">#&gt; D       16                abundances.csv        Jar_ID          ID</span>
<span class="co">#&gt; D       17                abundances.csv     Date_time       other</span>
<span class="co">#&gt; D       18                abundances.csv Lid_treatment   Treatment</span>
<span class="co">#&gt; D       19                abundances.csv      Jar_type       other</span>
<span class="co">#&gt; D       20                abundances.csv  count_number Measurement</span>
<span class="co">#&gt;                   mappingColumn      type description comment</span>
<span class="co">#&gt; D       1                  &lt;NA&gt; character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       2  oxygen concentration   numeric        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       3                  &lt;NA&gt; character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       4                  &lt;NA&gt; character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       5                  &lt;NA&gt; character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       6                  &lt;NA&gt;  datetime      ymdhms    &lt;NA&gt;</span>
<span class="co">#&gt; D       7         Lid_treatment character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       8                  &lt;NA&gt; character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       9                  &lt;NA&gt; character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       10            species_1 character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       11                smell character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       12                 &lt;NA&gt;  datetime      ymdhms    &lt;NA&gt;</span>
<span class="co">#&gt; D       13        Lid_treatment character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       14                 &lt;NA&gt; character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       15            species_3 character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       16                 &lt;NA&gt; character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       17                 &lt;NA&gt;  datetime      ymdhms    &lt;NA&gt;</span>
<span class="co">#&gt; D       18        Lid_treatment character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       19                 &lt;NA&gt; character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       20            abundance   numeric        &lt;NA&gt;    &lt;NA&gt;</span></pre></body></html></div>
</div>
<div id="validating-your-metadata" class="section level2">
<h2 class="hasAnchor">
<a href="#validating-your-metadata" class="anchor"></a>Validating your metadata</h2>
<p>The metadata in the spreadsheet can be validated by using the following command and which results in an object of class <code>dmdScheme_validation</code>.</p>
<div class="sourceCode" id="cb21"><html><body><pre class="r"><span class="fu"><a href="../reference/validate.html">validate</a></span>( <span class="fu"><a href="../reference/scheme_path_xlsx.html">scheme_path_xlsx</a></span>() )
<span class="co">#&gt; Validating Experiment</span>
<span class="co">#&gt; Validating Genus</span>
<span class="co">#&gt; Validating Treatments</span>
<span class="co">#&gt; Validating Measurement</span>
<span class="co">#&gt; Validating DataExtraction</span>
<span class="co">#&gt; Validating DataFileMetaData</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt; [1] NA</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#FF0000\"&gt;Overall MetaData - error&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "The result of the overall validation of the data."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] "The details contain the different validations of the metadata as a hierarchical list. errors propagate towards the root, i.e., if the 'worst' is a 'warning' in a validation in `details` the error here will be a 'warning' as well."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $structure</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt; [1] TRUE</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#00FF00\"&gt;Structural / Formal validity - OK&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "Test if the structure of the metadata is correct.  This includes column names, required info, ...  Should normally be OK, if no modification has been done."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Experiment</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt;   Module errorCode isOK</span>
<span class="co">#&gt; 1  types        OK TRUE</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#00FF00\"&gt;Experiment - OK&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "Test if the metadata concerning **Experiment** is correct.  This includes column names, required info, ... "</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] "The details are a table with one row per unique validation.\n The column `Module` contains the name of the validation,\n The column `error` contains the actual error of the validation.\n The following values are possible for the column `isTRUE`:\n \n    TRUE : If the validation was `OK`.\n    FALSE: If the validation was an `error`, `warning` or `note`.\n    NA   : If at least one v alidation resulted in `NA\n \n One or more FALSE or missing values values will result in an ERROR."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $types</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt;   name temperature light humidity incubator container microcosmVolume mediaType</span>
<span class="co">#&gt; 1 TRUE        TRUE  TRUE     TRUE      TRUE      TRUE            TRUE      TRUE</span>
<span class="co">#&gt;   mediaConcentration cultureConditions comunityType mediaAdditions duration</span>
<span class="co">#&gt; 1               TRUE              TRUE         TRUE           TRUE     TRUE</span>
<span class="co">#&gt;   comment</span>
<span class="co">#&gt; 1    &lt;NA&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#00FF00\"&gt;conversion of values into specified type lossless possible - OK&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "Test if the metadata entered follows the type for the column, i.e. integer, characterd, .... The validation is done by verifying if the column can be losslessly converted from character to the columnb type specified. the value NA is allowed in all column types, empty cells should be avoided."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] "The details are a table of the same dimension as the input (green) area of the meatadata sheet. The following values are possible:\n \n    FALSE: If the cell contains an error, i.e. can not be losslessly converted.\n    TRUE : If the cell can be losslessly converted and is OK.\n    NA   : empty cell\n \n One or more FALSE values will result in an ERROR."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Genus</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt;            Module errorCode  isOK</span>
<span class="co">#&gt; 1           types        OK  TRUE</span>
<span class="co">#&gt; 2 suggestedValues      note FALSE</span>
<span class="co">#&gt; 3   allowedValues        OK  TRUE</span>
<span class="co">#&gt; 4         IDField        OK  TRUE</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#55AA00\"&gt;Genus - note&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "Test if the metadata concerning ** Genus ** is correct.  This includes column names, required info, ... "</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] "The details are a table with one row per unique validation.\n The column `Module` contains the name of the validation,\n The column `error` contains the actual error of the validation.\n The following values are possible for the column `isTRUE`:\n \n    TRUE : If the validation was `OK`.\n    FALSE: If the validation was an `error`, `warning` or `note`.\n    NA   : If at least one v alidation resulted in `NA\n \n One or more FALSE or missing values values will result in an ERROR."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $types</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt;   speciesID colour density functionalGroup comment</span>
<span class="co">#&gt; 1      TRUE   TRUE    TRUE            TRUE    TRUE</span>
<span class="co">#&gt; 2      TRUE   TRUE    &lt;NA&gt;            TRUE    &lt;NA&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#00FF00\"&gt;conversion of values into specified type lossless possible - OK&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "Test if the metadata entered follows the type for the column, i.e. integer, characterd, .... The validation is done by verifying if the column can be losslessly converted from character to the columnb type specified. the value NA is allowed in all column types, empty cells should be avoided."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] "The details are a table of the same dimension as the input (green) area of the meatadata sheet. The following values are possible:\n \n    FALSE: If the cell contains an error, i.e. can not be losslessly converted.\n    TRUE : If the cell can be losslessly converted and is OK.\n    NA   : empty cell\n \n One or more FALSE values will result in an ERROR."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $suggestedValues</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt;   speciesID colour density functionalGroup comment</span>
<span class="co">#&gt; 1      &lt;NA&gt;   &lt;NA&gt;   FALSE            TRUE    &lt;NA&gt;</span>
<span class="co">#&gt; 2      &lt;NA&gt;   &lt;NA&gt;    TRUE           FALSE    &lt;NA&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#55AA00\"&gt;values in suggestedValues - note&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "Test if the metadata entered is ion the suggestedValues list. The value NA is allowed in all column types, empty cells should be avoided."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] "The details are a table of the same dimension as the input (green) area of the meatadata sheet. The following values are possible:\n \n    FALSE: If the cell value is not contained in the suggestedValues list.\n    TRUE : If the cell value is contained in the suggestedValues list.\n    NA   : empty cell or no suggested values specified\n \n One or more FALSE values will result in a WARNING."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $allowedValues</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt;   speciesID colour density functionalGroup comment</span>
<span class="co">#&gt; 1      &lt;NA&gt;   &lt;NA&gt;    &lt;NA&gt;            &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 2      &lt;NA&gt;   &lt;NA&gt;    &lt;NA&gt;            &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#00FF00\"&gt;values in allowedValues - OK&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "Test if the metadata entered is ion the allowedValues list. The value NA is allowed in all column types, empty cells should be avoided."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] "The details are a table of the same dimension as the input (green) area of the meatadata sheet. The following values are possible:\n \n    FALSE: If the cell value is not contained in the allowedValues list.\n    TRUE : If the cell value is contained in the allowedValues list.\n    NA   : empty cell or no allowed values specified\n \n One or more FALSE values will result in an ERROR."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $IDField</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt;                         hasIDField isOK</span>
<span class="co">#&gt; 1 tab has ID field in first column TRUE</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#00FF00\"&gt;ID Field present and in the first column - OK&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "Check if the tab's first column contains an ID field, named as `...ID`. This function does not check for uniqueness of this ID field!"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] "Returns a boolean value, with the following possible values:\n \n    TRUE  : The tab's first column is an ID field\n    FALSE : The tab's first column is not an ID field\n \n FALSE will result in an ERROR."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Treatments</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt;            Module errorCode  isOK</span>
<span class="co">#&gt; 1           types        OK  TRUE</span>
<span class="co">#&gt; 2 suggestedValues      note FALSE</span>
<span class="co">#&gt; 3   allowedValues        OK  TRUE</span>
<span class="co">#&gt; 4         IDField        OK  TRUE</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#55AA00\"&gt;Treatments - note&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "Test if the metadata concerning ** Treatments ** is correct.  This includes column names, required info, ... "</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] "The details are a table with one row per unique validation.\n The column `Module` contains the name of the validation,\n The column `error` contains the actual error of the validation.\n The following values are possible for the column `isTRUE`:\n \n    TRUE : If the validation was `OK`.\n    FALSE: If the validation was an `error`, `warning` or `note`.\n    NA   : If at least one v alidation resulted in `NA\n \n One or more FALSE or missing values values will result in an ERROR."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $types</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt;   treatmentID treatmentLevelHeight comment</span>
<span class="co">#&gt; 1        TRUE                 TRUE    &lt;NA&gt;</span>
<span class="co">#&gt; 2        TRUE                 TRUE    &lt;NA&gt;</span>
<span class="co">#&gt; 3        TRUE                 TRUE    &lt;NA&gt;</span>
<span class="co">#&gt; 4        TRUE                 TRUE    &lt;NA&gt;</span>
<span class="co">#&gt; 5        TRUE                 TRUE    &lt;NA&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#00FF00\"&gt;conversion of values into specified type lossless possible - OK&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "Test if the metadata entered follows the type for the column, i.e. integer, characterd, .... The validation is done by verifying if the column can be losslessly converted from character to the columnb type specified. the value NA is allowed in all column types, empty cells should be avoided."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] "The details are a table of the same dimension as the input (green) area of the meatadata sheet. The following values are possible:\n \n    FALSE: If the cell contains an error, i.e. can not be losslessly converted.\n    TRUE : If the cell can be losslessly converted and is OK.\n    NA   : empty cell\n \n One or more FALSE values will result in an ERROR."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $suggestedValues</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt;   treatmentID treatmentLevelHeight comment</span>
<span class="co">#&gt; 1       FALSE                FALSE    &lt;NA&gt;</span>
<span class="co">#&gt; 2       FALSE                FALSE    &lt;NA&gt;</span>
<span class="co">#&gt; 3       FALSE                FALSE    &lt;NA&gt;</span>
<span class="co">#&gt; 4       FALSE                FALSE    &lt;NA&gt;</span>
<span class="co">#&gt; 5       FALSE                FALSE    &lt;NA&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#55AA00\"&gt;values in suggestedValues - note&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "Test if the metadata entered is ion the suggestedValues list. The value NA is allowed in all column types, empty cells should be avoided."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] "The details are a table of the same dimension as the input (green) area of the meatadata sheet. The following values are possible:\n \n    FALSE: If the cell value is not contained in the suggestedValues list.\n    TRUE : If the cell value is contained in the suggestedValues list.\n    NA   : empty cell or no suggested values specified\n \n One or more FALSE values will result in a WARNING."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $allowedValues</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt;   treatmentID treatmentLevelHeight comment</span>
<span class="co">#&gt; 1        &lt;NA&gt;                 &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 2        &lt;NA&gt;                 &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 3        &lt;NA&gt;                 &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 4        &lt;NA&gt;                 &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 5        &lt;NA&gt;                 &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#00FF00\"&gt;values in allowedValues - OK&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "Test if the metadata entered is ion the allowedValues list. The value NA is allowed in all column types, empty cells should be avoided."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] "The details are a table of the same dimension as the input (green) area of the meatadata sheet. The following values are possible:\n \n    FALSE: If the cell value is not contained in the allowedValues list.\n    TRUE : If the cell value is contained in the allowedValues list.\n    NA   : empty cell or no allowed values specified\n \n One or more FALSE values will result in an ERROR."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $IDField</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt;                         hasIDField isOK</span>
<span class="co">#&gt; 1 tab has ID field in first column TRUE</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#00FF00\"&gt;ID Field present and in the first column - OK&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "Check if the tab's first column contains an ID field, named as `...ID`. This function does not check for uniqueness of this ID field!"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] "Returns a boolean value, with the following possible values:\n \n    TRUE  : The tab's first column is an ID field\n    FALSE : The tab's first column is not an ID field\n \n FALSE will result in an ERROR."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $Measurement</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt;            Module errorCode  isOK</span>
<span class="co">#&gt; 1           types        OK  TRUE</span>
<span class="co">#&gt; 2 suggestedValues      note FALSE</span>
<span class="co">#&gt; 3   allowedValues        OK  TRUE</span>
<span class="co">#&gt; 4         IDField        OK  TRUE</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#55AA00\"&gt;Measurement - note&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "Test if the metadata concerning ** Measurement ** is correct.  This includes column names, required info, ... "</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] "The details are a table with one row per unique validation.\n The column `Module` contains the name of the validation,\n The column `error` contains the actual error of the validation.\n The following values are possible for the column `isTRUE`:\n \n    TRUE : If the validation was `OK`.\n    FALSE: If the validation was an `error`, `warning` or `note`.\n    NA   : If at least one v alidation resulted in `NA\n \n One or more FALSE or missing values values will result in an ERROR."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $types</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt;   measurementID variable method unit object noOfSamplesInTimeSeries</span>
<span class="co">#&gt; 1          TRUE     TRUE   TRUE TRUE   TRUE                    TRUE</span>
<span class="co">#&gt; 2          TRUE     TRUE   TRUE TRUE   TRUE                    TRUE</span>
<span class="co">#&gt; 3          TRUE     TRUE   TRUE TRUE   TRUE                    TRUE</span>
<span class="co">#&gt; 4          TRUE     TRUE   TRUE TRUE   TRUE                    TRUE</span>
<span class="co">#&gt;   samplingVolume dataExtractionID measuredFrom comment</span>
<span class="co">#&gt; 1           TRUE             TRUE         TRUE    &lt;NA&gt;</span>
<span class="co">#&gt; 2           TRUE             TRUE         TRUE    &lt;NA&gt;</span>
<span class="co">#&gt; 3           TRUE             TRUE         TRUE    &lt;NA&gt;</span>
<span class="co">#&gt; 4           TRUE             TRUE         TRUE    &lt;NA&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#00FF00\"&gt;conversion of values into specified type lossless possible - OK&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "Test if the metadata entered follows the type for the column, i.e. integer, characterd, .... The validation is done by verifying if the column can be losslessly converted from character to the columnb type specified. the value NA is allowed in all column types, empty cells should be avoided."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] "The details are a table of the same dimension as the input (green) area of the meatadata sheet. The following values are possible:\n \n    FALSE: If the cell contains an error, i.e. can not be losslessly converted.\n    TRUE : If the cell can be losslessly converted and is OK.\n    NA   : empty cell\n \n One or more FALSE values will result in an ERROR."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $suggestedValues</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 1</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt;   measurementID variable method  unit object noOfSamplesInTimeSeries</span>
<span class="co">#&gt; 1          &lt;NA&gt;    FALSE   TRUE  TRUE   TRUE                    &lt;NA&gt;</span>
<span class="co">#&gt; 2          &lt;NA&gt;     TRUE  FALSE  TRUE   TRUE                    &lt;NA&gt;</span>
<span class="co">#&gt; 3          &lt;NA&gt;    FALSE  FALSE FALSE   TRUE                    &lt;NA&gt;</span>
<span class="co">#&gt; 4          &lt;NA&gt;     TRUE  FALSE FALSE  FALSE                    &lt;NA&gt;</span>
<span class="co">#&gt;   samplingVolume dataExtractionID measuredFrom comment</span>
<span class="co">#&gt; 1           &lt;NA&gt;             &lt;NA&gt;         &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 2           &lt;NA&gt;             &lt;NA&gt;         &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 3           &lt;NA&gt;             &lt;NA&gt;         &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 4           &lt;NA&gt;             &lt;NA&gt;         &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#55AA00\"&gt;values in suggestedValues - note&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "Test if the metadata entered is ion the suggestedValues list. The value NA is allowed in all column types, empty cells should be avoided."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] "The details are a table of the same dimension as the input (green) area of the meatadata sheet. The following values are possible:\n \n    FALSE: If the cell value is not contained in the suggestedValues list.\n    TRUE : If the cell value is contained in the suggestedValues list.\n    NA   : empty cell or no suggested values specified\n \n One or more FALSE values will result in a WARNING."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $allowedValues</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt;   measurementID variable method unit object noOfSamplesInTimeSeries</span>
<span class="co">#&gt; 1          &lt;NA&gt;     &lt;NA&gt;   &lt;NA&gt; &lt;NA&gt;   &lt;NA&gt;                    &lt;NA&gt;</span>
<span class="co">#&gt; 2          &lt;NA&gt;     &lt;NA&gt;   &lt;NA&gt; &lt;NA&gt;   &lt;NA&gt;                    &lt;NA&gt;</span>
<span class="co">#&gt; 3          &lt;NA&gt;     &lt;NA&gt;   &lt;NA&gt; &lt;NA&gt;   &lt;NA&gt;                    &lt;NA&gt;</span>
<span class="co">#&gt; 4          &lt;NA&gt;     &lt;NA&gt;   &lt;NA&gt; &lt;NA&gt;   &lt;NA&gt;                    &lt;NA&gt;</span>
<span class="co">#&gt;   samplingVolume dataExtractionID measuredFrom comment</span>
<span class="co">#&gt; 1           &lt;NA&gt;             &lt;NA&gt;         &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 2           &lt;NA&gt;             &lt;NA&gt;         &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 3           &lt;NA&gt;             &lt;NA&gt;         &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 4           &lt;NA&gt;             &lt;NA&gt;         &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#00FF00\"&gt;values in allowedValues - OK&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "Test if the metadata entered is ion the allowedValues list. The value NA is allowed in all column types, empty cells should be avoided."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] "The details are a table of the same dimension as the input (green) area of the meatadata sheet. The following values are possible:\n \n    FALSE: If the cell value is not contained in the allowedValues list.\n    TRUE : If the cell value is contained in the allowedValues list.\n    NA   : empty cell or no allowed values specified\n \n One or more FALSE values will result in an ERROR."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $IDField</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt;                         hasIDField isOK</span>
<span class="co">#&gt; 1 tab has ID field in first column TRUE</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#00FF00\"&gt;ID Field present and in the first column - OK&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "Check if the tab's first column contains an ID field, named as `...ID`. This function does not check for uniqueness of this ID field!"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] "Returns a boolean value, with the following possible values:\n \n    TRUE  : The tab's first column is an ID field\n    FALSE : The tab's first column is not an ID field\n \n FALSE will result in an ERROR."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $DataExtraction</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt;            Module errorCode isOK</span>
<span class="co">#&gt; 1           types        OK TRUE</span>
<span class="co">#&gt; 2 suggestedValues        OK TRUE</span>
<span class="co">#&gt; 3   allowedValues        OK TRUE</span>
<span class="co">#&gt; 4         IDField        OK TRUE</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#00FF00\"&gt;DataExtraction - OK&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "Test if the metadata concerning ** DataExtraction ** is correct.  This includes column names, required info, ... "</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] "The details are a table with one row per unique validation.\n The column `Module` contains the name of the validation,\n The column `error` contains the actual error of the validation.\n The following values are possible for the column `isTRUE`:\n \n    TRUE : If the validation was `OK`.\n    FALSE: If the validation was an `error`, `warning` or `note`.\n    NA   : If at least one v alidation resulted in `NA\n \n One or more FALSE or missing values values will result in an ERROR."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $types</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt;   dataExtractionID method parameter value comment</span>
<span class="co">#&gt; 1             TRUE   &lt;NA&gt;      &lt;NA&gt;  &lt;NA&gt;    TRUE</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#00FF00\"&gt;conversion of values into specified type lossless possible - OK&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "Test if the metadata entered follows the type for the column, i.e. integer, characterd, .... The validation is done by verifying if the column can be losslessly converted from character to the columnb type specified. the value NA is allowed in all column types, empty cells should be avoided."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] "The details are a table of the same dimension as the input (green) area of the meatadata sheet. The following values are possible:\n \n    FALSE: If the cell contains an error, i.e. can not be losslessly converted.\n    TRUE : If the cell can be losslessly converted and is OK.\n    NA   : empty cell\n \n One or more FALSE values will result in an ERROR."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $suggestedValues</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt;   dataExtractionID method parameter value comment</span>
<span class="co">#&gt; 1             &lt;NA&gt;   TRUE      &lt;NA&gt;  &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#00FF00\"&gt;values in suggestedValues - OK&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "Test if the metadata entered is ion the suggestedValues list. The value NA is allowed in all column types, empty cells should be avoided."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] "The details are a table of the same dimension as the input (green) area of the meatadata sheet. The following values are possible:\n \n    FALSE: If the cell value is not contained in the suggestedValues list.\n    TRUE : If the cell value is contained in the suggestedValues list.\n    NA   : empty cell or no suggested values specified\n \n One or more FALSE values will result in a WARNING."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $allowedValues</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt;   dataExtractionID method parameter value comment</span>
<span class="co">#&gt; 1             &lt;NA&gt;   &lt;NA&gt;      &lt;NA&gt;  &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#00FF00\"&gt;values in allowedValues - OK&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "Test if the metadata entered is ion the allowedValues list. The value NA is allowed in all column types, empty cells should be avoided."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] "The details are a table of the same dimension as the input (green) area of the meatadata sheet. The following values are possible:\n \n    FALSE: If the cell value is not contained in the allowedValues list.\n    TRUE : If the cell value is contained in the allowedValues list.\n    NA   : empty cell or no allowed values specified\n \n One or more FALSE values will result in an ERROR."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $IDField</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt;                         hasIDField isOK</span>
<span class="co">#&gt; 1 tab has ID field in first column TRUE</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#00FF00\"&gt;ID Field present and in the first column - OK&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "Check if the tab's first column contains an ID field, named as `...ID`. This function does not check for uniqueness of this ID field!"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] "Returns a boolean value, with the following possible values:\n \n    TRUE  : The tab's first column is an ID field\n    FALSE : The tab's first column is not an ID field\n \n FALSE will result in an ERROR."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $DataFileMetaData</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt;           Module errorCode  isOK</span>
<span class="co">#&gt; 1          types        OK  TRUE</span>
<span class="co">#&gt; 2  allowedValues        OK  TRUE</span>
<span class="co">#&gt; 3 dataFilesExist     error FALSE</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#FF0000\"&gt;DataFileMetaData - error&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "Test if the metadata concerning **DataExtraction** is correct.  This includes column names, required info, ... "</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] "The details are a table with one row per unique validation.\n The column `Module` contains the name of the validation,\n The column `error` contains the actual error of the validation.\n The following values are possible for the column `isTRUE`:\n \n    TRUE : If the validation was `OK`.\n    FALSE: If the validation was an `error`, `warning` or `note`.\n    NA   : If at least one v alidation resulted in `NA\n \n One or more FALSE or missing values values will result in an ERROR."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $types</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt;    dataFileName columnName columnData mappingColumn type description comment</span>
<span class="co">#&gt; 1          TRUE       TRUE       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 2          TRUE       TRUE       TRUE          TRUE TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 3          TRUE       TRUE       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 4          TRUE       TRUE       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 5          TRUE       TRUE       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 6          TRUE       TRUE       TRUE          &lt;NA&gt; TRUE        TRUE    &lt;NA&gt;</span>
<span class="co">#&gt; 7          TRUE       TRUE       TRUE          TRUE TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 8          TRUE       TRUE       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 9          TRUE       TRUE       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 10         TRUE       &lt;NA&gt;       TRUE          TRUE TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 11         TRUE       TRUE       TRUE          TRUE TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 12         TRUE       TRUE       TRUE          &lt;NA&gt; TRUE        TRUE    &lt;NA&gt;</span>
<span class="co">#&gt; 13         TRUE       TRUE       TRUE          TRUE TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 14         TRUE       TRUE       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 15         TRUE       &lt;NA&gt;       TRUE          TRUE TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 16         TRUE       TRUE       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 17         TRUE       TRUE       TRUE          &lt;NA&gt; TRUE        TRUE    &lt;NA&gt;</span>
<span class="co">#&gt; 18         TRUE       TRUE       TRUE          TRUE TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 19         TRUE       TRUE       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 20         TRUE       TRUE       TRUE          TRUE TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#00FF00\"&gt;conversion of values into specified type lossless possible - OK&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "Test if the metadata entered follows the type for the column, i.e. integer, characterd, .... The validation is done by verifying if the column can be losslessly converted from character to the columnb type specified. the value NA is allowed in all column types, empty cells should be avoided."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] "The details are a table of the same dimension as the input (green) area of the meatadata sheet. The following values are possible:\n \n    FALSE: If the cell contains an error, i.e. can not be losslessly converted.\n    TRUE : If the cell can be losslessly converted and is OK.\n    NA   : empty cell\n \n One or more FALSE values will result in an ERROR."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $allowedValues</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt;    dataFileName columnName columnData mappingColumn type description comment</span>
<span class="co">#&gt; 1          &lt;NA&gt;       &lt;NA&gt;       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 2          &lt;NA&gt;       &lt;NA&gt;       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 3          &lt;NA&gt;       &lt;NA&gt;       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 4          &lt;NA&gt;       &lt;NA&gt;       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 5          &lt;NA&gt;       &lt;NA&gt;       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 6          &lt;NA&gt;       &lt;NA&gt;       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 7          &lt;NA&gt;       &lt;NA&gt;       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 8          &lt;NA&gt;       &lt;NA&gt;       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 9          &lt;NA&gt;       &lt;NA&gt;       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 10         &lt;NA&gt;       &lt;NA&gt;       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 11         &lt;NA&gt;       &lt;NA&gt;       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 12         &lt;NA&gt;       &lt;NA&gt;       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 13         &lt;NA&gt;       &lt;NA&gt;       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 14         &lt;NA&gt;       &lt;NA&gt;       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 15         &lt;NA&gt;       &lt;NA&gt;       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 16         &lt;NA&gt;       &lt;NA&gt;       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 17         &lt;NA&gt;       &lt;NA&gt;       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 18         &lt;NA&gt;       &lt;NA&gt;       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 19         &lt;NA&gt;       &lt;NA&gt;       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; 20         &lt;NA&gt;       &lt;NA&gt;       TRUE          &lt;NA&gt; TRUE        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#00FF00\"&gt;values in allowedValues - OK&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "Test if the metadata entered is ion the allowedValues list. The value NA is allowed in all column types, empty cells should be avoided."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] "The details are a table of the same dimension as the input (green) area of the meatadata sheet. The following values are possible:\n \n    FALSE: If the cell value is not contained in the allowedValues list.\n    TRUE : If the cell value is contained in the allowedValues list.\n    NA   : empty cell or no allowed values specified\n \n One or more FALSE values will result in an ERROR."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $dataFilesExist</span>
<span class="co">#&gt; $error</span>
<span class="co">#&gt; [1] 3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $details</span>
<span class="co">#&gt;                    dataFileName  IsOK</span>
<span class="co">#&gt; 1 dissolved_oxygen_measures.csv FALSE</span>
<span class="co">#&gt; 2                     smell.csv FALSE</span>
<span class="co">#&gt; 3                abundances.csv FALSE</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $header</span>
<span class="co">#&gt; [1] "**&lt;span style=\"color:#FF0000\"&gt;`dataFile` exists in path - error&lt;/span&gt;**"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $description</span>
<span class="co">#&gt; [1] "Test if all `dataFile` exist in the given `path`. The `error` can have the following values apart from `OK`:\n \n    error   : If not all `dataFile` exist in the given `path`\n \n"</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $descriptionDetails</span>
<span class="co">#&gt; [1] "The details are a table with one row per unique `variable` The following values are possible for the column `isTRUE`:\n \n    TRUE : If `dataFile` exist in the given `path`\n    FALSE: If `dataFile` does not exist in the given `path`\n    NA   : empty cell\n \n One or more FALSE or missing values will result in an ERROR."</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $comment</span>
<span class="co">#&gt; [1] ""</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"                </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,"class")</span>
<span class="co">#&gt; [1] "dmdScheme_validation" "list"</span></pre></body></html></div>
<p>To create a report (html, pdf or docx) yo can use the <code><a href="../reference/report.html">report()</a></code> function:</p>
<div class="sourceCode" id="cb22"><html><body><pre class="r"><span class="fu"><a href="../reference/report.html">report</a></span>( <span class="fu"><a href="../reference/scheme_path_xlsx.html">scheme_path_xlsx</a></span>() )</pre></body></html></div>
<p>which will open a html report in your browser.</p>
</div>
<div id="converting-metadata-to-xml" class="section level2">
<h2 class="hasAnchor">
<a href="#converting-metadata-to-xml" class="anchor"></a>Converting Metadata to xml</h2>
<p>You can export the data to an an <code>xml_document</code> object as defined in the <code>xml2</code> package to do further processing:</p>
<div class="sourceCode" id="cb23"><html><body><pre class="r"><span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/read_excel.html">read_excel</a></span>( <span class="fu"><a href="../reference/scheme_path_xlsx.html">scheme_path_xlsx</a></span>() )
<span class="no">xml</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/as_xml.html">as_xml</a></span>( <span class="no">x</span>, <span class="kw">file</span> <span class="kw">=</span> <span class="no">xmlFile</span>)
<span class="no">xml</span>
<span class="co">#&gt; {xml_document}</span>
<span class="co">#&gt; &lt;dmdScheme fileName="dmdScheme_0.9.5.xlsx" dmdSchemeName="dmdScheme" dmdSchemeVersion="0.9.5" propertyName="dmdScheme" output="metadata"&gt;</span>
<span class="co">#&gt; [1] &lt;ExperimentList&gt;\n  &lt;Experiment&gt;\n    &lt;name&gt;ASR-expt1&lt;/name&gt;\n    &lt;temper ...</span>
<span class="co">#&gt; [2] &lt;GenusList&gt;\n  &lt;Genus speciesID="tt_1"&gt;\n    &lt;speciesID&gt;tt_1&lt;/speciesID&gt;\ ...</span>
<span class="co">#&gt; [3] &lt;TreatmentsList&gt;\n  &lt;Treatments treatmentID="Lid_treatment"&gt;\n    &lt;treatm ...</span>
<span class="co">#&gt; [4] &lt;MeasurementList&gt;\n  &lt;Measurement measurementID="oxygen concentration"&gt;\n ...</span>
<span class="co">#&gt; [5] &lt;DataExtractionList&gt;\n  &lt;DataExtraction dataExtractionID="Mol_Analy_pipel ...</span>
<span class="co">#&gt; [6] &lt;DataFileMetaDataList&gt;\n  &lt;DataFileMetaData&gt;\n    &lt;dataFileName&gt;dissolved ...</span></pre></body></html></div>
<p>You can save the xml to a file by using</p>
<div class="sourceCode" id="cb24"><html><body><pre class="r"><span class="no">xmlFile</span> <span class="kw">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempfile</a></span>(<span class="kw">fileext</span> <span class="kw">=</span> <span class="st">".xml"</span>)

<span class="kw pkg">xml2</span><span class="kw ns">::</span><span class="fu"><a href="http://xml2.r-lib.org/reference/write_xml.html">write_xml</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="no">xml</span>, <span class="kw">file</span> <span class="kw">=</span> <span class="no">xmlFile</span>)
<span class="co">#&gt; NULL</span></pre></body></html></div>
<p>or directly in one command</p>
<div class="sourceCode" id="cb25"><html><body><pre class="r"><span class="no">xmlFile</span> <span class="kw">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html">tempfile</a></span>(<span class="kw">fileext</span> <span class="kw">=</span> <span class="st">".xml"</span>)

<span class="fu"><a href="../reference/write_xml.html">write_xml</a></span>( <span class="kw">x</span> <span class="kw">=</span> <span class="no">x</span>, <span class="kw">file</span> <span class="kw">=</span> <span class="no">xmlFile</span> )
<span class="co">#&gt; [1] "/var/folders/50/wcr5bjwn75q595n6x82gxj280000gq/T//RtmpXFEJYP/file556d208f8ab.xml"</span></pre></body></html></div>
</div>
<div id="re-import-from-xml-into-r" class="section level2">
<h2 class="hasAnchor">
<a href="#re-import-from-xml-into-r" class="anchor"></a>Re-import from xml into R</h2>
<div class="sourceCode" id="cb26"><html><body><pre class="r"><span class="no">x</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/read_xml.html">read_xml</a></span>( <span class="no">xmlFile</span> )
<span class="no">x</span>
<span class="co">#&gt;  dmdScheme - dmdSchemeSet</span>
<span class="co">#&gt;      Experiment - dmdSchemeData</span>
<span class="co">#&gt; A       Names : name | temperature | light | humidity | incubator | container | microcosmVolume | mediaType | mediaConcentration | cultureConditions | comunityType | mediaAdditions | duration | comment</span>
<span class="co">#&gt; A       Units : NA | NA | NA | NA | NA | NA | ml | NA | g/l | NA | NA | NA | days | NA</span>
<span class="co">#&gt; A       Type  : character | character | character | character | character | character | numeric | character | numeric | character | character | character | integer | character</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;                name temperature        light humidity      incubator</span>
<span class="co">#&gt; D       1 ASR-expt1          20 semi-ambient  ambient not given here</span>
<span class="co">#&gt;                                    container microcosmVolume mediaType</span>
<span class="co">#&gt; D       1 Duran type bottle, red lids, 250ml             100       PPM</span>
<span class="co">#&gt;           mediaConcentration cultureConditions      comunityType</span>
<span class="co">#&gt; D       1               0.55             dirty initially unknown</span>
<span class="co">#&gt;                                                                   mediaAdditions</span>
<span class="co">#&gt; D       1 Wheat seeds added on specific dates, see file wheat_seed_additions.csv</span>
<span class="co">#&gt;           duration comment</span>
<span class="co">#&gt; D       1      100    &lt;NA&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;      Genus - dmdSchemeData</span>
<span class="co">#&gt; A       Names : speciesID | colour | density | functionalGroup | comment</span>
<span class="co">#&gt; A       Units : NA | NA | cells / ml | NA | NA</span>
<span class="co">#&gt; A       Type  : character | character | character | character | character</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;           speciesID  colour density functionalGroup</span>
<span class="co">#&gt; D       1      tt_1    ATCC       1     bacterivore</span>
<span class="co">#&gt; D       2   unknown unknown    &lt;NA&gt;         unknown</span>
<span class="co">#&gt;                                                            comment</span>
<span class="co">#&gt; D       1 http://www.lgcstandards-atcc.org/products/all/30007.aspx</span>
<span class="co">#&gt; D       2                                                     &lt;NA&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;      Treatments - dmdSchemeData</span>
<span class="co">#&gt; A       Names : treatmentID | treatmentLevelHeight | comment</span>
<span class="co">#&gt; A       Units : NA | NA | NA</span>
<span class="co">#&gt; A       Type  : character | character | character</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;             treatmentID treatmentLevelHeight comment</span>
<span class="co">#&gt; D       1 Lid_treatment                Loose    &lt;NA&gt;</span>
<span class="co">#&gt; D       2 Lid_treatment                Tight    &lt;NA&gt;</span>
<span class="co">#&gt; D       3     species_1        tt_1, unknown    &lt;NA&gt;</span>
<span class="co">#&gt; D       4     species_2              unknown    &lt;NA&gt;</span>
<span class="co">#&gt; D       5     species_3                 tt_1    &lt;NA&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;      Measurement - dmdSchemeData</span>
<span class="co">#&gt; A       Names : measurementID | variable | method | unit | object | noOfSamplesInTimeSeries | samplingVolume | dataExtractionID | measuredFrom | comment</span>
<span class="co">#&gt; A       Units : NA | NA | NA | NA | NA | NA | ml | NA | NA | NA</span>
<span class="co">#&gt; A       Type  : character | character | character | character | character | integer | numeric | character | character | character</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;                  measurementID  variable         method               unit</span>
<span class="co">#&gt; D       1 oxygen concentration        DO presens Optode                  %</span>
<span class="co">#&gt; D       2            abundance abundance      molecular              count</span>
<span class="co">#&gt; D       3                smell     smell           nose rotten eggs or not</span>
<span class="co">#&gt; D       4         sequenceData       DNA            NGS         Nucleotide</span>
<span class="co">#&gt;                 object noOfSamplesInTimeSeries samplingVolume</span>
<span class="co">#&gt; D       1    community                      50            0.0</span>
<span class="co">#&gt; D       2      species                       6            0.5</span>
<span class="co">#&gt; D       3    community                       6            0.0</span>
<span class="co">#&gt; D       4 DNA fragment                       6            0.0</span>
<span class="co">#&gt;              dataExtractionID measuredFrom comment</span>
<span class="co">#&gt; D       1                none          raw    &lt;NA&gt;</span>
<span class="co">#&gt; D       2 Mol_Analy_pipeline1 sequenceData    &lt;NA&gt;</span>
<span class="co">#&gt; D       3                none          raw    &lt;NA&gt;</span>
<span class="co">#&gt; D       4                none          raw    &lt;NA&gt;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;      DataExtraction - dmdSchemeData</span>
<span class="co">#&gt; A       Names : dataExtractionID | method | parameter | value | comment</span>
<span class="co">#&gt; A       Units : NA | NA | NA | NA | NA</span>
<span class="co">#&gt; A       Type  : character | character | character | character | character</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;              dataExtractionID method parameter value</span>
<span class="co">#&gt; D       1 Mol_Analy_pipeline1   &lt;NA&gt;      &lt;NA&gt;  &lt;NA&gt;</span>
<span class="co">#&gt;                                   comment</span>
<span class="co">#&gt; D       1 See description in file xxx.yyy</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;      DataFileMetaData - dmdSchemeData</span>
<span class="co">#&gt; A       Names : dataFileName | columnName | columnData | mappingColumn | type | description | comment</span>
<span class="co">#&gt; A       Units : NA | NA | NA | NA | NA | NA | NA</span>
<span class="co">#&gt; A       Type  : character | character | character | character | character | character | character</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;                             dataFileName    columnName  columnData</span>
<span class="co">#&gt; D       1  dissolved_oxygen_measures.csv        Jar_ID          ID</span>
<span class="co">#&gt; D       2  dissolved_oxygen_measures.csv            DO Measurement</span>
<span class="co">#&gt; D       3  dissolved_oxygen_measures.csv        Unit_1       other</span>
<span class="co">#&gt; D       4  dissolved_oxygen_measures.csv          Mode       other</span>
<span class="co">#&gt; D       5  dissolved_oxygen_measures.csv      Location       other</span>
<span class="co">#&gt; D       6  dissolved_oxygen_measures.csv     Date_time       other</span>
<span class="co">#&gt; D       7  dissolved_oxygen_measures.csv Lid_treatment   Treatment</span>
<span class="co">#&gt; D       8  dissolved_oxygen_measures.csv      Jar_type       other</span>
<span class="co">#&gt; D       9  dissolved_oxygen_measures.csv        Jar_ID          ID</span>
<span class="co">#&gt; D       10                     smell.csv          &lt;NA&gt;     Species</span>
<span class="co">#&gt; D       11                     smell.csv         smell Measurement</span>
<span class="co">#&gt; D       12                     smell.csv          Date       other</span>
<span class="co">#&gt; D       13                     smell.csv Lid_treatment   Treatment</span>
<span class="co">#&gt; D       14                     smell.csv      Jar_type       other</span>
<span class="co">#&gt; D       15                abundances.csv          &lt;NA&gt;     Species</span>
<span class="co">#&gt; D       16                abundances.csv        Jar_ID          ID</span>
<span class="co">#&gt; D       17                abundances.csv     Date_time       other</span>
<span class="co">#&gt; D       18                abundances.csv Lid_treatment   Treatment</span>
<span class="co">#&gt; D       19                abundances.csv      Jar_type       other</span>
<span class="co">#&gt; D       20                abundances.csv  count_number Measurement</span>
<span class="co">#&gt;                   mappingColumn      type description comment</span>
<span class="co">#&gt; D       1                  &lt;NA&gt; character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       2  oxygen concentration   numeric        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       3                  &lt;NA&gt; character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       4                  &lt;NA&gt; character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       5                  &lt;NA&gt; character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       6                  &lt;NA&gt;  datetime      ymdhms    &lt;NA&gt;</span>
<span class="co">#&gt; D       7         Lid_treatment character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       8                  &lt;NA&gt; character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       9                  &lt;NA&gt; character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       10            species_1 character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       11                smell character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       12                 &lt;NA&gt;  datetime      ymdhms    &lt;NA&gt;</span>
<span class="co">#&gt; D       13        Lid_treatment character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       14                 &lt;NA&gt; character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       15            species_3 character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       16                 &lt;NA&gt; character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       17                 &lt;NA&gt;  datetime      ymdhms    &lt;NA&gt;</span>
<span class="co">#&gt; D       18        Lid_treatment character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       19                 &lt;NA&gt; character        &lt;NA&gt;    &lt;NA&gt;</span>
<span class="co">#&gt; D       20            abundance   numeric        &lt;NA&gt;    &lt;NA&gt;</span></pre></body></html></div>
</div>
<div id="general-things" class="section level2">
<h2 class="hasAnchor">
<a href="#general-things" class="anchor"></a>General things</h2>
<p>The <code>dmdScheme</code> data in R can be also saved, loaded, and edited using the standard R commands.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Rainer M. Krug.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
